src/app/app.component.ts

```diff
@@ -58,11 +59,52 @@ export class AppComponent {
   templateUrl: './modal/modal.component.html',
   styleUrls: ['./modal/modal.component.css']
 })
-export class AppComponentModal {
+export class AppComponentModal implements OnInit {
+  mylist = [];
+  reach_reference: reach;
+  ini_mass: number;
+  ini_time: number;
+  new_reach: reach;
+  output: {};
 
   constructor(
     public dialogRef: MatDialogRef<AppComponentModal>,
```

Comment:
I think the issue is with how you're calling the modal. I made it work by moving all the functions from the modal.component.ts to here where you have the AppComponentModal implemented. 

I think a better fix would be instead of implementing the component here, you should create a function to show the modal in modal.component.ts, usually I see that involving Observables and Subjects, so you'd set the subject on click instead of having this separate dialogRef here. I can show you how it's done in other projects.